src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -726,6 +738,31 @@ public ProjectManager(Action<Exception>? exceptionLogger, Action<string, Message
             IEnumerable<Uri> projectFiles,
             ProjectInformation.Loader projectLoader,
             Func<Uri, FileContentManager?>? openInEditor = null)
+        {
+            return this.LoadProjectsAsync(this.userProjects, projectFiles, projectLoader, openInEditor);
+        }
+
+        /// <summary>
+        /// To be used whenever a temporary project file is added.
+        /// Note that by calling this routine, all processing will be blocked until loading has finished...
+        /// </summary>
+        public Task LoadTemporaryProjectAsync(
```

Comment:
@rmshaffer  Before we merge this, since we don't want to keep this in the public API:
```suggestion
        private Task LoadTemporaryProjectAsync(
```

Reply:
Edit: I saw that the language server calls into this. That's rather unfortunate. It would be good to mark it such that people don't use it - as awkward as it is, I'd suggest marking it with an Obsolete attribute anyway.

Reply:
Is there a way to correct the public API surface, then, so that we don't have to resort to that? If I understood our discussions correctly, `ProjectManager` needs to have _some_ way of identifying these temporary projects so that it can treat them distinctly. And so no matter what we choose, there will be _some_ public contract between `LanguageServer`, which creates those temporary projects, and `ProjectManager`, which loads them.

Or am I misunderstanding - is your idea that we should ideally treat the temporary projects no differently from ordinary projects?